packages/jaeger-ui/src/types/trace.tsx

```diff
@@ -78,6 +78,11 @@ export type Trace = TraceData & {
   // Number of orphan spans (spans referencing parent spans that don't exist in the trace)
   orphanSpanCount?: number;
 
+  // Optimized data structures - created once during trace transformation
+  spanMap: Map<string, Span>;
+  tree: TreeNode<string>;
+  nodesBySpanId: Map<string, TreeNode<string>>;
```

Comment:
I changed my mind about this. TreeNode type does not provide any additional benefits if we already have childSpanIds on the Span type. I would like to do this instead:
  1. Keep `spanMap: Map<string, Span>`
  2. Replace `tree` with `rootSpans: Array<Span>`
  3. on the Span type, replace `childSpanIds` with `childSpans:; Array<Span>`
  4. Update the callers to reflect these changes.